src/games/supported/Adventure.cpp

```diff
@@ -0,0 +1,122 @@
+/* *****************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ * *****************************************************************************
+ * A.L.E (Arcade Learning Environment)
+ * Copyright (c) 2009-2013 by Yavar Naddaf, Joel Veness, Marc G. Bellemare and 
+ *   the Reinforcement Learning and Artificial Intelligence Laboratory
+ * Released under the GNU General Public License; see License.txt for details. 
+ *
+ * Based on: Stella  --  "An Atari 2600 VCS Emulator"
+ * Copyright (c) 1995-2007 by Bradford W. Mott and the Stella team
+ *
+ * *****************************************************************************
+ */
+#include "Adventure.hpp"
+
+#include "../RomUtils.hpp"
+
+AdventureSettings::AdventureSettings() {
+
+  reset();
+}
+
+/* create a new instance of the rom */
+RomSettings* AdventureSettings::clone() const {
+
+  RomSettings* rval = new AdventureSettings();
+  *rval = *this;
+  return rval;
+}
+
+/* process the latest information from ALE */
+void AdventureSettings::step(const System& system) {
+
+  int chalice_status = readRam(&system, 0xB9);
+  bool chalice_in_yellow_castle = chalice_status == 0x12;
+
+  if (chalice_in_yellow_castle) {
+    m_reward = 1;
```

Comment:
Hm, not that it matters, but I don't quite understand you. There is no score in this game, but that doesn't mean there should be no reward.

In that respect it's more like chess or go, where you are only rewarded at the end, but not with a point (unless you play in a tournament). That was also my motivation for adding the -1; anyrhing else is undecided, and I guess inconclusive games (I don't know the details of the timeout that must exist, is that external to ALE or is there a step limit?) are then draws.

So, aside from the fact that you do say that we should leave it in, under what circumstance could we ever use a game that nevee hands out any reward at all? No, you don't need to answer, it's not that important, just wondering.